tasmota/berry/drivers/Core200S.be

```diff
@@ -0,0 +1,712 @@
+import string
+import webserver
+
+class StateEnum
+    var value_map
+    var value
+
+    def init(value_map, value)
+        self.value_map = value_map
+        if type(value) == 'int'
+            for k : self.value_map.keys()
+                if self.value_map[k] == value
+                    self.value = value
+                    break
+                end
+            end
+        elif type(value) == 'string'
+            if self.value_map.contains(value)
+                self.value = self.value_map[value]
+            end
+        end
+
+        if self.value == nil
+            raise "StateEnumError", "Invalid type or value"
+        end
+    end
+
+    def tostring()
+        for k : self.value_map.keys()
+            if self.value_map[k] == self.value
+                return k
+            end
+        end
+        return "invalid"
+    end
+
+    def toint()
+        return self.value
+    end
+
+    def ==(other)
+        return self.value == other.value
+    end
+
+end
+
+class OnOffState : StateEnum
+    static ON = OnOffState("on")
+    static OFF = OnOffState("off")
+    def init(value)
+        super(self).init({"off": 0, "on": 1}, value)
+    end
+end
+
+class FanMode : OnOffState
+    def init(value)
+        super(self).init(value)
+    end
+end
+
+class ChildLockMode : OnOffState
+    def init(value)
+        super(self).init(value)
+    end
+end
+
+class SleepMode : OnOffState
+    def init(value)
+        super(self).init(value)
+    end
+end
+
+class DisplayAutoOffMode : StateEnum
+    static OFF = DisplayAutoOffMode("off")
+    static ON = DisplayAutoOffMode("on")
+    def init(value)
+        super(self).init({"off": 1, "on": 0}, value)
+    end
+end
+
+class DisplayMode : StateEnum
+    def init(value)
+        super(self).init({"off": 0, "on": 100}, value)
+    end
+end
+
+class FanSpeedMode : StateEnum
+    static LOW = FanSpeedMode("low")
+    static MED = FanSpeedMode("med")
+    static FULL = FanSpeedMode("full")
+
+    def init(value)
+        super(self).init({"low": 1, "med": 2, "full": 3}, value)
+    end
+end
+
+class NightLightMode : StateEnum
+    static OFF = NightLightMode("off")
+    static LOW = NightLightMode("low")
+    static FULL = NightLightMode("full")
+    def init(value)
+        super(self).init({"off": 0, "low": 50, "full": 100}, value)
+    end
+end
+
+def modulo(num, mod)
+    return ((num % mod) + mod) % mod
+end
+
+def info_c2s(msg)
+    log("C2S: " + msg, 2)
+end
+
+def log_c2s(msg)
```

Comment:
This does the same as `info_c2s`

Reply:
ah yeah, actually i wanted to set a different log level here :-D